frontend/components/ManageStakingPage/StakingContractSection/index.tsx

```diff
@@ -101,91 +131,124 @@ export const StakingContractSection = ({
-      return (
-        <CountdownUntilMigration
-          activeStakingContractInfo={activeStakingContractInfo}
-        />
-      );
+    if (!hasEnoughOlasToMigrate) {
+      return {
+        canMigrate: false,
+        reason: <Text>Insufficient OLAS to migrate</Text>,
+      };
+    }
+    // stopped service requirements
+    if (
+      [
+        DeploymentStatus.DEPLOYED,
+        DeploymentStatus.DEPLOYING,
+        DeploymentStatus.STOPPING,
+      ].some((status) => status === serviceStatus) // allow nil serviceStatus to pass, so first-timers can migrate
+    ) {
+      return {
+        canMigrate: false,
+        reason: <Text>Pearl is currently running</Text>,
+      };
     }
 
-    if (!hasEnoughOlasToMigrate) {
-      return `Insufficient OLAS to migrate, ${minimumOlasRequiredToMigrate} OLAS required in total.`;
+    // active staking program requirements
+    if (!isStakingContractInfoLoaded) {
+      return {
+        canMigrate: false,
+        reason: <Text>Loading staking contract information...</Text>,
+      };
     }
 
-    // App version compatibility not implemented yet
-    // if (!isAppVersionCompatible) {
-    //   return 'Pearl update required to migrate';
-    // }
+    // user is not actively staked, allow migration
+    if (activeStakingProgram === null) {
+      return { canMigrate: true };
+    }
 
-    if (serviceStatus === DeploymentStatus.DEPLOYED) {
-      return 'Pearl is currently running, turn it off before switching';
+    // USER IS STAKED
+    if (!isServiceStaked) {
+      return {
+        canMigrate: false,
+        reason: <Text>Service is not staked</Text>,
+      };
     }
 
-    if (serviceStatus === DeploymentStatus.DEPLOYING) {
-      return 'Pearl is currently deploying, please turn it off before switching';
+    if (!activeStakingProgramMeta?.canMigrateTo.includes(stakingProgram)) {
+      return {
+        canMigrate: false,
+        reason: 'Migration not supported for this contract',
+      };
     }
 
-    if (serviceStatus === DeploymentStatus.STOPPING) {
-      return 'Pearl is currently stopping, please wait before switching';
+    if (activeStakingContractInfo && !isServiceStakedForMinimumDuration) {
+      return {
+        canMigrate: false,
+        reason: (
+          <CountdownUntilMigration
+            activeStakingContractInfo={activeStakingContractInfo}
+          />
+        ),
+      };
     }
+
+    return { canMigrate: true };
   }, [
     activeStakingContractInfo,
+    activeStakingProgram,
     activeStakingProgramMeta?.canMigrateTo,
     hasEnoughOlasToMigrate,
     hasEnoughRewards,
     hasEnoughSlots,
     isBalanceLoaded,
-    isSelected,
+    isSelectedOrUnstakedDefault,
+    isServiceStaked,
     isServiceStakedForMinimumDuration,
-    minimumOlasRequiredToMigrate,
+    isStakingContractInfoLoaded,
     serviceStatus,
     stakingProgram,
   ]);
+  // further conditions for migration from an actively staked service
+  !isServiceStaked ||
```

Comment:
confused what this does? not ifs, vars, etc?

Reply:
left by accident, thanks for catching